astropy/coordinates/representation.py

```diff
@@ -2222,6 +2222,62 @@ def to_cartesian(self):
         return CartesianRepresentation(x=x, y=y, z=z, copy=False)
 
 
+class WGS84GeodeticRepresentation(BaseRepresentation):
+    """
+    Representation of points in 3D geodetic (WGS84) coordinates.
+
+    Parameters
+    ----------
+    lon, lat : `~astropy.units.Quantity`
+        The longitude and latitude of the point(s), in angular units. The
+        latitude should be between -90 and 90 degrees, and the longitude will
+        be wrapped to an angle between 0 and 360 degrees. These can also be
+        instances of `~astropy.coordinates.Angle`,
+        `~astropy.coordinates.Longitude`, or `~astropy.coordinates.Latitude`.
+
+    height : `~astropy.units.Quantity`
+        The height to the point(s).
+
+    copy : bool, optional
+        If `True` (default), arrays will be copied. If `False`, arrays will
+        be references, though possibly broadcast to ensure matching shapes.
+
+    """
+
+    attr_classes = {'lon': Longitude,
+                    'lat': Latitude,
+                    'height': u.Quantity}
+
+    def __init__(self, lon, lat=None, height=None, copy=True):
+        super().__init__(lon, lat, height, copy=copy)
+
+    def to_cartesian(self):
+        """
+        Converts WGS84 geodetic coordinates to 3D rectangular (geocentric)
+        cartesian coordiantes.
+        """
+        # TODO: Check ret?
+        xyz, ret = erfa_ufunc.gd2gc(1,
+                         self.lon.to_value(u.radian),
+                         self.lat.to_value(u.radian),
+                         self.height.to_value(u.m))
+        return CartesianRepresentation(xyz, unit=u.m, xyz_axis=-1, copy=False)
+
+    @classmethod
+    def from_cartesian(cls, cart):
+        """
+        Converts 3D rectangular cartesian coordinates (assumed geocentric) to
+        WGS84 geodetic coordinates.
+        """
+        # TODO: Check ret?
+        lon, lat, height, ret = erfa_ufunc.gc2gd(1, cart.get_xyz(xyz_axis=-1).to_value(u.m))
+        return cls(
+            Longitude(lon * u.radian, u.degree,
```

Comment:
While you are at it: do change this to `lon << u.radian`, etc., that avoids an extra copy.

Reply:
Good catch!